### PR TITLE
Add google-cloud-container_analysis library

### PR DESCRIPTION
**Pull request by dazuma**
Initial synth of `google-cloud-container_analysis`. The synth file for this library includes some hackery to move namespaces and file paths around because the proto path is `google/devtools/containeranalysis` whereas we want the final path to be `google/cloud/container_analysis`.

Also includes Kokoro configs for the library.

### CODE REVIEW

**Comment by blowmage** [comment]
Curious why the name `google-cloud-container_analysis` and the path `google/cloud/container_analysis` is preferred to `google-cloud-containeranalysis` and `google/cloud/containeranalysis`. I can see why `Google::Cloud::ContainerAnalysis` is preferable to `Google::Cloud::Containeranalysis`, but why force the underscore in the gem and file path?

**Comment by dazuma** [comment]
@blowmage 
1. It's consistent with the usual Ruby convention of `CamelCase` <=> `snake_case`
2. It's consistent with what we've done for all our other libraries (`video_intelligence`, `resource_manager`, etc.)

Are you thinking we might consider removing the underscore?

**Comment by blowmage** [comment]
I would prefer no underscores in the situations that would match things like existing proto file paths. There are plenty of places where Ruby does not use the Rails inflections. Calling `require "securerandom"` loads `SecureRandom`. Calling `require "stringio"` loads `StringIO`. Heck, even rails gems like `activerecord` load `ActiveRecord`. 

I would also prefer to keep existing gem names and paths for libs like `bigquery` and `pubsub`, while defining constants like `BigQuery` and `PubSub`.

**Comment by blowmage** [comment]
BTW, I am aware this is not 100% consistent with the [Gem Naming Guide](https://guides.rubygems.org/name-your-gem/), but I think that the existence of prior art in the proto file paths and maintaining consistency with them should have some influence on the naming decisions.

File: google-cloud-container_analysis/.rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+AllCops:
+  TargetRubyVersion: 2.2
+  Exclude:
+    - "google-cloud-container_analysis.gemspec"
+    - "lib/google/**/*"
+    - "Rakefile"
+    - "test/**/*"
+    - "acceptance/**/*"
```
Comment (quartzmo) [inline google-cloud-container_analysis/.rubocop.yml]: Please add: `TargetRubyVersion: 2.2`

**Comment by dazuma** [comment]
@blowmage I see. The issue I was trying to fix was that the proto class namespace is being set to `Google::Cloud::ContainerAnalysis::V1` (https://github.com/googleapis/googleapis/blob/master/google/devtools/containeranalysis/v1/containeranalysis.proto#L30) whereas the proto path is actually `google/devtools/containeranalysis/v1`. But you're right, our prior art does not have us modifying the file paths. (e.g. pubsub: the proto class namespace is `Google::Cloud::PubSub` whereas the file path remains `google/pubsub`.) I'll revert the file moves.

File: google-cloud-container_analysis/.rubocop.yml (hunk 1 above)
Comment (dazuma) [reply google-cloud-container_analysis/.rubocop.yml]: It's already there. :)

**Comment by dazuma** [comment]
Updated.

**Pull request merged**